library/core/src/marker.rs

```diff
@@ -823,7 +823,7 @@ unsafe impl<T: ?Sized> Freeze for &mut T {}
 /// [`pin` module]: crate::pin
 #[stable(feature = "pin", since = "1.33.0")]
 #[rustc_on_unimplemented(
-    note = "consider using `Box::pin`",
+    note = "consider using `core::pin::pin!`\nconsider using `Box::pin` if you want ownership of the pinned value",
```

Comment:
What's the policy about mentioning `core` in an diagnostic message? Should we mention the `pin!` macro instead?

I'm not convinced about the "if you want ownership of the pinned value" part. The important difference between the two is the local storage. What about: "if you need to access the pinned value outside of the current scope"?

Reply:
Yeah, I suppose the std::pin::pin! macro would be better

Also yeah, that message works better, it captures the "local pin" semantics far better. I'll make the changes and commit